x/globalfee/ante/fee_utils.go

```diff
@@ -139,3 +74,35 @@ func Find(coins sdk.Coins, denom string) (bool, sdk.Coin) {
 		}
 	}
 }
+
+// splitCoinsByDenoms returns the given coins split in two whether
+// their demon is or isn't found in the given denom map.
+func splitCoinsByDenoms(feeCoins sdk.Coins, denomMap map[string]bool) (feeCoinsNonZeroDenom sdk.Coins, feeCoinsZeroDenom sdk.Coins) {
+	for _, fc := range feeCoins {
+		_, found := denomMap[fc.Denom]
+		if found {
+			feeCoinsZeroDenom = append(feeCoinsZeroDenom, fc)
+		} else {
+			feeCoinsNonZeroDenom = append(feeCoinsNonZeroDenom, fc)
+		}
+	}
+
+	return feeCoinsNonZeroDenom.Sort(), feeCoinsZeroDenom.Sort()
+}
+
+// splitFees returns the given fees nonzero coins
+// and a map storing the zero coins's denoms
+func splitFees(fees sdk.Coins) (sdk.Coins, map[string]bool) {
```

Comment:
!Naming `splitFees` should have a more descriptive name such as `getNonZeroFees`